sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -626,74 +617,165 @@ func withSuperpowered(superpowered bool) oci.SpecOpts {
 		// Set the `control_t` process label so the host container can
 		// interact with the API and modify its local state files.
 		return oci.Compose(
-			seccomp.WithDefaultProfile(),
 			oci.WithSelinuxLabel("system_u:system_r:control_t:s0"),
 		)
 	}
 
 	return oci.Compose(
 		oci.WithHostNamespace(runtimespec.PIDNamespace),
 		oci.WithParentCgroupDevices,
-		oci.WithPrivileged,
 		oci.WithNewPrivileges,
 		oci.WithSelinuxLabel("system_u:system_r:super_t:s0"),
-		oci.WithAllDevicesAllowed,
 	)
 }
 
-// withRootFileSystemMounts adds container options to mount the root filesystem
-func withRootFilesystemMounts(mountRootFilesystem bool) oci.SpecOpts {
-	// if mountRootFilesystem, return a no-op SpecOpts function
-	if !mountRootFilesystem {
-		return func(_ context.Context, _ oci.Client, _ *containers.Container, s *runtimespec.Spec) error {
-			return nil
-		}
+// withMounts configures all the mounts for the container. If the contianer is privileged,
+// the following mounts from the underlying host are included:
```

Comment:
As above, I think this is better separated into multiple functional options that each have no logic (or a minimal amount of logic) inside.  One that handles default mounts and another that deals with privileged mounts.